### ISSUE TITLE
[OverflowList] Add separatorRenderer prop

### ISSUE THREAD

**Issue by cmslewis** [palantir/blueprint#2905]
#### Environment

- __Package version(s)__: 3.4.0

#### Feature request

Love the `<OverflowList>`. I'm hitting a snag though: currently, to render separators between items, I need to use CSS.

```scss
.my-overflow-list-visible-item {
  &:before {
    content: ",";
  }
}
```

This is how the docs example works, for instance.

I wonder if there could be a simple `separatorRenderer` prop to allow consumers to insert richer separators—or just basic ones—easily.

#### Examples

If we move forward with this, we may want to let users configure when to include a separator before the "overflow item." For instance, my use case was a plain-English list of values that needed to be collapsed into an "overflow item" if it exceeded the available width of the container (where clicking the overflow item would open a `<Popover>` with the remaining values):

> `// has visible items, no overflow item, has separators`
> __Values:__ A, B, C 
> `// has visible items, has overflow item, has separators`
> __Values:__ A, B, and <a href="#">1 other</a>
> `// no visible items, has overflow item, no separators`
> __Values:__ <a href="#">3 values</a>
> `// has visible item, has overflow item, no separators`
> __Values:__ A and 1 other

But that might be biting off more than the component should chew.

**Comment by giladgray** [comment]
yep i thought of this complexity too. how about `separatorRenderer` and `showOverflowSeparator`?

**Comment by giladgray** [comment]
we also need to add an `onOverflow` callback to observe the overflow state. this came up in #2802.
